pkg/kapp/config/default.go

```diff
@@ -582,6 +586,16 @@ changeRuleBindings:
       - notMatcher:
           matcher: *disableDefaultChangeGroupAnnMatcher
 
+- rules:
+  - "delete before deleting change-groups.kapp.k14s.io/serviceaccount"
+  resourceMatchers:
+  - apiVersionKindMatcher: {kind: App, apiVersion: kappctrl.k14s.io/v1alpha1}
+
+- rules:
+  - "delete before deleting change-groups.kapp.k14s.io/rbac"
+  resourceMatchers:
+  - apiVersionKindMatcher: {kind: App, apiVersion: kappctrl.k14s.io/v1alpha1}
```

Comment:
- we should combine it with above rule. note that rules is an array.
- this should also be marked as `ignoreIfCyclical: true` since it's not a hard requirement and user may want to override it
- this should not just apply to App, but also to PackageInstall

Reply:
For packageInstall, I have tested with current changes it is working as we want. I have added ordered output above in a comment.
https://github.com/vmware-tanzu/carvel-kapp/pull/552#issuecomment-1178823844